src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala

```diff
@@ -257,48 +258,43 @@ class ModeChoiceMultinomialLogit(val beamServices: BeamServices, val model: Mult
 
 object ModeChoiceMultinomialLogit {
 
-  def buildModelFromConfig(mnlConfig: Agents.ModalBehaviors.MulitnomialLogit): MultinomialLogit = {
-    val mnlData: Vector[MnlData] = Vector(
-      new MnlData("COMMON", "cost", "multiplier", -1.0),
-      new MnlData("car", "intercept", "intercept", mnlConfig.params.car_intercept),
-      new MnlData("cav", "intercept", "intercept", mnlConfig.params.cav_intercept),
-      new MnlData("walk", "intercept", "intercept", mnlConfig.params.walk_intercept),
-      new MnlData(
-        "ride_hail",
-        "intercept",
-        "intercept",
-        mnlConfig.params.ride_hail_intercept
-      ),
-      new MnlData(
-        "ride_hail_pooled",
-        "intercept",
-        "intercept",
-        mnlConfig.params.ride_hail_pooled_intercept
+  def buildModelFromConfig(mnlConfig: Agents.ModalBehaviors.MulitnomialLogit): MultinomialLogit[String, String] = {
+
+    val commonUtility: Map[String, UtilityFunctionOperation] = Map(
+      "cost" -> UtilityFunctionOperation("multiplier", -1)
+    )
+
+    val mnlUtilityFunctions: Map[String, Map[String, UtilityFunctionOperation]] = Map(
+      "car" -> Map(
+        "intercept" ->
+        UtilityFunctionOperation("intercept", mnlConfig.params.car_intercept)
       ),
-      new MnlData("bike", "intercept", "intercept", mnlConfig.params.bike_intercept),
-      new MnlData(
-        "walk_transit",
-        "intercept",
-        "intercept",
-        mnlConfig.params.walk_transit_intercept
+      "cav"       -> Map("intercept" -> UtilityFunctionOperation("intercept", mnlConfig.params.cav_intercept)),
+      "walk"      -> Map("intercept" -> UtilityFunctionOperation("intercept", mnlConfig.params.walk_intercept)),
+      "ride_hail" -> Map("intercept" -> UtilityFunctionOperation("intercept", mnlConfig.params.ride_hail_intercept)),
+      "walk"      -> Map("intercept" -> UtilityFunctionOperation("intercept", mnlConfig.params.walk_intercept)),
```

Comment:
This is a duplicate, and I don't see anything that is missing from the original - so maybe just remove this?